publisher/provider.py

```diff
@@ -0,0 +1,21 @@
+from dataclasses import dataclass
+from typing import List, Optional, Protocol
+
+Symbol = str
+
+
+@dataclass
+class Price:
+    price: float
+    conf: float
+
+
+class Provider(Protocol):
```

Comment:
Maybe consider making this an [abstract base class](https://realpython.com/python-interface/#using-abstract-method-declaration)?

Reply:
yeah i agree abc is better on this use case. will make a PR for it.